src/model/AssetsCompiler.ts

```diff
@@ -376,41 +404,127 @@ export class AssetsCompiler {
+                        .toLowerCase()}`
+                );
+
+                this.addProjectDependenciesRecursively(
+                    repositoryDir,
+                    projects,
+                    pluginDescriptor.repoName,
+                    pluginDescriptor.name,
+                    pluginPathInNodeModules,
+                    true,
+                    runConfig
+                );
+            } else {
+                const pluginPath = this.findPluginPath(
+                    repositoryDir,
+                    pluginDescriptor.name,
+                    AssetsCompiler.knownRepoDependencies
+                );
+                this.addProjectDependenciesRecursively(
+                    repositoryDir,
+                    projects,
+                    pluginDescriptor.repoName,
+                    pluginDescriptor.name,
+                    pluginPath,
+                    false,
+                    runConfig
+                );
+            }
+        });
+    }
+
+    private static checkIfPluginExistsAsNpmArtifact(
+        repositoryDir: string,
+        pluginDescriptor: PluginDescriptor
+    ): void {
+        // validate that the npm package for the repository exists in the node_modules
+        const pathToRepoInNodeModules = path.resolve(
+            repositoryDir,
+            'node_modules',
+            '@cplace-assets',
+            `${pluginDescriptor.repoName}`
+        );
+        if (!fs.existsSync(pathToRepoInNodeModules)) {
+            throw Error(
+                `[${pluginDescriptor.repoName}] repository npm package should be resolved in node_modules, but it's missing`
             );
-            this.addProjectDependenciesRecursively(
-                repositoryDir,
-                projects,
-                repoDependencies,
-                pluginDescriptor.name,
-                pluginPath,
-                runConfig
+        }
+
+        // check in the package.json file of the repo npm package if the plugin is a npm dependency (if the plugin assets wer published)
+        const packageJsonPath = path.resolve(
+            pathToRepoInNodeModules,
+            'package.json'
+        );
+        const packageJson = JSON.parse(
+            fs.readFileSync(packageJsonPath, 'utf8')
+        );
+
+        const expectedPluginPackageName = `@cplace-assets/${
+            pluginDescriptor.repoName
+        }_${pluginDescriptor.name.replaceAll('.', '-').toLowerCase()}`;
+        const expectedPathToPluginInNodeModules = path.resolve(
+            repositoryDir,
+            'node_modules',
+            expectedPluginPackageName
+        );
+
+        if (
+            packageJson.dependencies &&
+            packageJson.dependencies[expectedPluginPackageName] &&
+            !fs.existsSync(expectedPathToPluginInNodeModules)
+        ) {
+            throw Error(
+                `[${pluginDescriptor.name}] plugin npm package should be resolved in node_modules, but it's missing`
```

Comment:
If this fails what could I do to fix the issue? Maybe a hint in the error message could help.